R/scATAC.Peak.annotation.R

```diff
@@ -275,18 +290,15 @@ peaks_on_gene <- function(peak_features,annotations=NULL, gene_element=NULL, spl
   cores <- as.numeric(availableCores() -2)
   cat("available_cores:", cores, "\n")
   plan(multisession,workers = cores )
-  peak_list <- list()
   computing <- cores*1000
-  data_iterat <- nrow(peaks)%/%computing
-  for (n in 1:data_iterat) {
-    peak_list[[n]] <- peaks[(n*computing-(computing-1)):(n*computing),]
-  }
-  if ((nrow(peaks) %% computing) != 0) {
-    peak_list[[length(peak_list) +1 ]] <- peaks[(length(peak_list)*computing+1):nrow(peaks),]}
+  peak_list <- create_data_chunks(peaks, computing)
```

Comment:
create_data_chunks splis the order, of the matrix. I would suggest leaving it like you suggested initially [6a20e4f](https://github.com/PatZeis/scATAC.Peak.annotation/pull/2/commits/6a20e4f8a93f8a0e7bdedd19532c45e99889f6f4), with accounting for the condition where less peaks are present then size of the chunks.

Reply:
I am not sure rows are scrambled. This can happen if a factor is provided, but I am using a sequence. In any case, is it important to keep the order?

Reply:
also by providing a sequence the order of the matrix is changed. But essentially the order is not important, but would require some additional changes for some of the functions

Reply:
also by providing a sequence the order is changed. But the order is not really important, but would just require some additional changes in some of the functions